packages/tx/src/baseTransaction.ts

```diff
@@ -370,7 +370,9 @@ export abstract class BaseTransaction<T extends TransactionType>
       const chainIdBigInt = bytesToBigInt(toBytes(chainId))
       if (common) {
         if (common.chainId() !== chainIdBigInt) {
-          const msg = this._errorMsg('The chain ID does not match the chain ID of Common')
+          const msg = this._errorMsg(
+            `The chain ID of ${chainIdBigInt} does not match the chain ID of Common (${common.chainId()})`
```

Comment:
In these cases we always tried to keep the original string as before for backwards compatibility reasons (if someone does string comparison, and put the extra info at the end (with „(expected: x, actual: y)“ or similar e.g.).

Think that’s a good practice and we should stick to it.

Reply:
(So this is not a 100% thing, but it helps if someone e.g. does:)
If „chain ID does not match“ in errorMsg do something

Reply:
Ah ok I see! I'll update. I will (as a kind of "chore task") go over all our error messages to see if we can improve the error strings (at the end)

Reply:
Thanks, it’s a bit of a „blurry“ rule, but I remember that people complained in the past when we changed error msgs too aggressively.